lib/parser.js

```diff
@@ -341,7 +341,14 @@ module.exports = function (args, opts, y18n) {
       var configPath = argv[configKey] || configLookup[configKey]
       if (configPath) {
         try {
-          var config = require(path.resolve(process.cwd(), configPath))
+          var config = null
+          var resolvedConfigPath = path.resolve(process.cwd(), configPath)
+
+          if (typeof flags.configs[configKey] === 'function') {
+            config = flags.configs[configKey](resolvedConfigPath)
+          } else {
+            config = require(resolvedConfigPath)
```

Comment:
I recall that someone switched it to require so that they could have `.js` config files -- I think we should hold off on this change, in `v4` perhaps we direct people to provide a custom require hook that uses `require()`.